codec/enforceproto/extraneous_fields.go

```diff
@@ -0,0 +1,200 @@
+package enforceproto
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+
+	gogoproto "github.com/gogo/protobuf/proto"
+	"google.golang.org/protobuf/encoding/protowire"
+	"google.golang.org/protobuf/proto"
+)
+
+const bit11Reserved = 1 << 10
+
+// CheckExtraneousFieldsProto walks through the protobuf serialized bytes in b, and tries to
+// compare its fields against those in msg. The error returned if non-nil will contain
+// the listing of the extraneous fields by tagNumber and wireType.
+func CheckExtraneousFieldsProto(b []byte, msg proto.Message) error {
```

Comment:
We're not using `google.golang.org/protobuf/proto` at all so I'm not sure there's value in supporting it here. My general preference is to not support things we don't use.

Reply:
Gotcha, thanks, I've removed it.